src/main/java/org/jenkinsci/plugins/docker/workflow/FromFingerprintStep.java

```diff
@@ -87,8 +89,9 @@ public String getToolName() {
             String fromImage = null;
             FilePath dockerfile = workspace.child(step.dockerfile);
             InputStream is = dockerfile.read();
+            BufferedReader r = null;
```

Comment:
BTW I recommend the simpler (immutable local variable) idiom

```java
BufferedReader r = new BufferedReader(new InputStreamReader(is, "ISO-8859-1"));
try {
    // ..
} finally {
    r.close();
}
```

which can be mechanically converted in `-source 7` to the even better

```java
try (BufferedReader r = new BufferedReader(new InputStreamReader(is, "ISO-8859-1"))) {
    // ..
}
```

`IOUtils.closeQuietly` is just useless. Actually this block as written is now _worse_ than the original, since a `BufferedReader` does not consume any resources other than heap, whereas a `FileInputStream` or the like does, and we are no longer conclusively closing `is`—very likely we are, but it depends on two other constructors not throwing `RuntimeException`/`Error`. The best idiom in `-source 6` is to always nest blocks:

```java
InputStream is = dockerfile.read();
try {
    BufferedReader r = new BufferedReader(new InputStreamReader(is, "ISO-8859-1"));
    try {
        // ..
    } finally {
        r.close();
    }
} finally {
    is.close();
}
```

which again in Java 7 can be mechanically translated to the ideal form (in terms of brevity, safety, and diagnosability):

```java
try (InputStream is = dockerfile.read();
     InputStreamReader isr = new InputStreamReader(is, "ISO-8859-1");
     BufferedReader r = new BufferedReader(isr)) {
    // ..
}
```